network/src/connectivity_manager/mod.rs

```diff
@@ -285,20 +296,27 @@ where
             .filter(|peer_id| !eligible.contains_key(peer_id))
             .cloned()
             .collect();
-        for p in stale_connections.into_iter() {
-            info!(
-                "{} Should no longer be connected to peer: {}",
-                self.network_context,
-                p.short_str()
-            );
-            // Close existing connection.
-            if let Err(e) = self.connection_reqs_tx.disconnect_peer(p).await {
+
+        if !stale_connections.is_empty() {
```

Comment:
should we instead check this and return early? excessive nesting makes me sad

Reply:
Yeah, can do.